librz/bin/p/bin_z64.c

```diff
@@ -80,13 +79,16 @@ static bool check_buffer(RzBuffer *b) {
 }
 
 static bool load_buffer(RzBinFile *bf, RzBinObject *obj, RzBuffer *b, Sdb *sdb) {
-	if (check_buffer(b)) {
-		ut8 buf[sizeof(N64Header)] = { 0 };
-		rz_buf_read_at(b, 0, buf, sizeof(buf));
-		obj->bin_obj = memcpy(&n64_header, buf, sizeof(N64Header));
-		return true;
+	N64Header *hdr = RZ_NEW0(N64Header);
+	if (!check_buffer(b)) {
+		free(hdr);
+		return false;
 	}
-	return false;
+	ut8 buf[sizeof(N64Header)] = { 0 };
+	rz_buf_read_at(b, 0, buf, sizeof(buf));
+	memcpy(hdr, buf, sizeof(N64Header));
```

Comment:
You can directly read into the header, no need to first create a buffer on the stack and then copy it. You just need to cast the pointer.

Reply:
As mentioned before the old code is wrong and bad. According to the old code the whole binary is big endian, so you should parse the header like the other plug-ins. I shared before some links to the nin3ds parser which uses the offset api of RzBuffer. Please do the same here.